### ISSUE TITLE
Dataset with multiple shapes with null dimensions cannot be assigned 2d matrix

### ISSUE THREAD

**Issue by mcafaro** [NeurodataWithoutBorders/matnwb#21]
ElectricalSeries.data is defined by the spec as:
```yaml
    name: data
    shape:
    - - null
    - - null
      - null
```
Attempting to assign a 2d matrix to data on the generated class causes an error:
```matlab
>> es = types.ElectricalSeries;
>> es.data = ones(2)
Error using types.ElectricalSeries/validate_data (line 65)
ElectricalSeries.data: val must have [1,2] dimensions

Error in types.TimeSeries/set.data (line 99)
      obj.data = validate_data(obj, val);
```
**Failing test:**
```matlab
>> nwbtest('ProcedureName', 'testWriteDatasetWithMultipleNullShapes')
```

**Comment by ln-vidrio** [comment]
Fixed with 418f8e6180c096aba530320c956dfd2820598090

**Comment by ln-vidrio** [comment]
new test under
`>> nwbtest('ProcedureName', 'testMultipleNullShapesDataset')`
